### ISSUE TITLE
Navigation is broken if slides are generated from a JavaScript block

### ISSUE THREAD

**Issue by cimi** [FormidableLabs/spectacle#169]
Navigation breaks when reaching slides generated by a JS block. Specifically:

* I can't move forward into the first generated slide, using neither mouse or keyboard (arrows and space)
* if I enter the presentation directly on the second slide (`localhost:3000/1`) I can move forward and back, but only once. Regardless of direction after the first transition it's stuck again.

I am able to reproduce this on a fresh checkout of `spectacle-boilerplate`. This works fine:

```
<Spectacle theme={theme}>
  <Deck transition={["zoom", "slide"]} transitionDuration={500}>
    <Slide bgColor="primary">
      <Heading size={1} fit caps lineHeight={1} textColor="black">Before</Heading>
    </Slide>
    <Slide bgColor="tertiary">
      <Heading size={1} fit caps lineHeight={1} textColor="black">Generated slide 1</Heading>
    </Slide>
    <Slide bgColor="tertiary">
      <Heading size={1} fit caps lineHeight={1} textColor="black">Generated slide 2</Heading>
    </Slide>
    <Slide bgColor="primary">
      <Heading size={1} fit caps lineHeight={1} textColor="black">After</Heading>
    </Slide>
  </Deck>
</Spectacle>
```

This is broken, having all the symptoms described above:

```
<Spectacle theme={theme}>
  <Deck transition={["zoom", "slide"]} transitionDuration={500}>
    <Slide transition={["zoom"]} bgColor="primary">
      <Heading size={1} fit caps lineHeight={1} textColor="black">Before</Heading>
    </Slide>
    {[1, 2, 3, 4, 5].map((num) => {
      return (<Slide transition={["slide"]} bgColor="tertiary">
        <Heading size={1} fit caps lineHeight={1} textColor="black">{`Generated slide ${num}`}</Heading>
      </Slide>);
    })}
    <Slide transition={["zoom"]} bgColor="primary">
      <Heading size={1} fit caps lineHeight={1} textColor="black">After</Heading>
    </Slide>
  </Deck>
</Spectacle>
```

I guess it shouldn't matter, but here my system specs:

```
➜  ~ node -v
v4.2.2
➜  ~ npm -v
3.8.8
➜  ~ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.10.5
BuildVersion:	14F1713
```

**Comment by kenwheeler** [comment]
Fixed, sorry about that.

**Comment by cimi** [comment]
Sorry to resurrect this after so much time, but there's still a problem here. I seem to be able to reproduce the initial problem on a fresh checkout of the boilerplate (using Spectacle 1.0.7).

I had tested when you initially fixed and it did work in my presentation. However, my presentation broke when I wanted to generate 14 slides programatically, instead of 9.

I've tried adding a `key={}` prop to the generated slide to get rid of the warning in the browser console but that doesn't help with the navigation breaking.
